qiskit_experiments/calibration/analysis/fine_amplitude_analysis.py

```diff
@@ -0,0 +1,156 @@
+
+    @classmethod
+    def _default_options(cls):
+        """Return the default analysis options.
+
+        See :meth:`~qiskit_experiment.analysis.CurveAnalysis._default_options` for
+        descriptions of analysis options.
+        """
+        default_options = super()._default_options()
+        default_options.p0 = {"amp": None, "d_theta": None, "phase": None, "baseline": None}
+        default_options.bounds = {"amp": None, "d_theta": None, "phase": None, "baseline": None}
+        default_options.fit_reports = {"d_theta": "d_theta"}
+        default_options.xlabel = "Number of gates (n)"
+        default_options.ylabel = "Population"
+        default_options.angle_per_gate = None
+        default_options.phase_offset = None
+        default_options.number_guesses = 101
+
+        return default_options
+
+    def _setup_fitting(self, **options) -> Union[Dict[str, Any], List[Dict[str, Any]]]:
+        """Fitter options."""
+        user_p0 = self._get_option("p0")
+        user_bounds = self._get_option("bounds")
+        n_guesses = self._get_option("number_guesses")
+
+        max_y, min_y = np.max(self._data().y), np.min(self._data().y)
+        b_guess = (max_y + min_y) / 2
+        a_guess = max_y - min_y
+
+        max_abs_y = np.max(np.abs(self._data().y))
+
+        # Base the initial guess on the intended angle_per_gate.
+        angle_per_gate = self._get_option("angle_per_gate")
+
+        if angle_per_gate is None:
+            raise CalibrationError("The angle_per_gate was not specified in the analysis options.")
+
+        guess_range = max(abs(angle_per_gate), np.pi / 2)
+
+        fit_options = []
+
+        for angle in np.linspace(-guess_range, guess_range, n_guesses):
+            fit_option = {
+                "p0": {
+                    "amp": user_p0["amp"] or a_guess,
+                    "d_theta": angle,
+                    "baseline": b_guess,
+                },
+                "bounds": {
+                    "amp": user_bounds.get("amp", None) or (-2 * max_abs_y, 2 * max_abs_y),
+                    "d_theta": user_bounds.get("d_theta", None) or (-np.pi, np.pi),
+                    "baseline": user_bounds.get("d_theta", None) or (-1 * max_abs_y, 1 * max_abs_y),
+                },
+            }
+
+            fit_options.append(fit_option)
+
+        return fit_options
+
+    def _post_analysis(self, analysis_result: CurveAnalysisResult) -> CurveAnalysisResult:
+        """Algorithmic criteria for whether the fit is good or bad.
+
+        A good fit has:
+            - a reduced chi-squared lower than three,
+            - a measured angle error that is smaller than the angle_per_gate.
+        """
+        fit_d_theta = get_opt_value(analysis_result, "d_theta")
+        angle_per_gate = self._get_option("angle_per_gate")
+
+        criteria = [
+            analysis_result["reduced_chisq"] < 3,
+            abs(fit_d_theta) < abs(angle_per_gate),

```

Comment:
Following on from the other discussion about the possibility of angle_per_gate being zero, maybe we need a different reference here than `angle_per_gate`?

Reply:
Maybe this could be an analysis_option with a default value of pi/2?

Reply:
See here: f09ae8c